packages/linter/src/rules/ImagesHaveAltText.ts

```diff
@@ -3,16 +3,31 @@ import { Rule } from "../rule";
 
 export class ImagesHaveAltText extends Rule {
   run({ $ }: Context) {
-    let imgsWithoutAlt = "";
+    let imgsWithoutAlt: { [key: string]: number } = {};
 
     $("amp-img").each(function (i, elem) {
       if (!elem.attribs.alt) {
-        imgsWithoutAlt = imgsWithoutAlt + "- " + elem.attribs.src + "\n";
+        if (typeof imgsWithoutAlt[elem.attribs.src] == "undefined") {
+          imgsWithoutAlt[elem.attribs.src] = 1;
+        } else {
+          imgsWithoutAlt[elem.attribs.src] += 1;
+        }
       }
     });
 
-    return imgsWithoutAlt.length > 0
-      ? this.warn(`Missing alt text from images: \n` + imgsWithoutAlt)
+    let output = "";
+    let m = `${JSON.stringify(imgsWithoutAlt)}`.split(",");
+    m.forEach(function (el) {
+      output +=
+        el
+          .replace('{"', "")
+          .replace('":', " (found ")
+          .replace('"', "")
+          .replace("}", "") + " times)\n";
+    });
+
+    return Object.keys(imgsWithoutAlt).length > 0
+      ? this.warn("Missing alt text from images: \n" + output)
```

Comment:
Shouldn't we fail?

Reply:
I wanted to. But according to the feedback I've gotten it should be a warning since the Story could still be valid AMP and surfaced. Its mostly to stay aligned with the official AMP test tool.